tests/unittests/BackendTest.cpp

```diff
@@ -188,6 +190,67 @@ TEST_P(BackendTest, CompileWithoutConstants) {
   auto function = backend->compile(F, opts);
 }
 
+/// Test that the runtimeBundle includes only symbols from its function and not
+/// the whole module.
+TEST_P(BackendTest, BundleFunctionSymbolsOnly) {
+  Module mod;
+  PlaceholderBindings bindings;
+  Function *F = mod.createFunction("main");
+  auto *X = mod.createConstant(ElemKind::FloatTy, {3}, "Y");
+  X->getHandle() = {1., 2., 3.};
+  auto *pow = F->createPow("Pow1", X, 2.0);
+  auto *save = F->createSave("save", pow);
+  bindings.allocate(save->getPlaceholder());
+  PlaceholderBindings bindings2;
+  Function *F2 = mod.createFunction("main2");
+  auto *X2 = mod.createConstant(ElemKind::FloatTy, {3}, "X2");
+  X2->getHandle() = {1., 2., 3.};
+  auto *pow2 = F2->createPow("Pow2", X2, 2.0);
+  auto *save2 = F2->createSave("save2", pow2);
+  bindings2.allocate(save2->getPlaceholder());
+
+  std::unique_ptr<Backend> backend(createBackend(GetParam()));
+  CompilationOptions opts;
+  auto function = backend->compile(F, opts);
```

Comment:
nit: Using opts only on one `compile()` call -- can probably remove it and just use `compile(F)` on both. Same for other test below.